homeassistant/components/recorder/migration.py

```diff
@@ -1414,6 +1401,12 @@ def _apply_update(self) -> None:
         _drop_index(self.session_maker, "events", "ix_events_event_type_time_fired")
         _drop_index(self.session_maker, "states", "ix_states_last_updated")
         _drop_index(self.session_maker, "events", "ix_events_time_fired")
+        with session_scope(session=self.session_maker()) as session:
+            # In version 31 we migrated all the time_fired, last_updated, and last_changed
+            # columns to be timestamps. In version 32 we need to wipe the old columns
+            # since they are no longer used and take up a significant amount of space.
+            assert self.instance.engine is not None, "engine should never be None"
+            _wipe_old_string_time_columns(self.instance, self.instance.engine, session)
```

Comment:
This is limited to 100k rows for MySQL / MariaDB / PostgreSQL. Should we instead loop until there are no more rows?

Reply:
I figured it wasn't urgent to delete them and it was better to let them get cleaned up over time so the migration didn't take too long.  At this point since most people have been migrated it might make more sense to do the full cleanup.

Reply:
OK, that makes sense 👍